mobile/src/main/java/ch/epfl/sdp/mobile/ui/features/profile/ProfileStateScreen.kt

```diff
@@ -0,0 +1,154 @@
+      )
+    }
+    items(state.matches) { match ->
+      val title = strings.profileMatchTitle(match.adversary)
+      val subtitle = chooseSubtitle(match.matchResult, match.numberOfMoves)
+      Match(title, subtitle, PawniesIcons.SectionSocial)
+    }
+  }
+}
+
+@Composable
+fun chooseSubtitle(matchResult: MatchResult, nMoves: Int): String {
+  val strings = LocalLocalizedStrings.current
+  return when (matchResult) {
+    is Win ->
+        strings.profileMatchInfo(matchResult.toString(), matchResult.reason.toString(), nMoves)
+    is Loss ->
+        strings.profileMatchInfo(matchResult.toString(), matchResult.reason.toString(), nMoves)
+    is Tie -> strings.profileTieInfo(nMoves)
+  }
+}
+
+/**
+ * Composes the profile header given the profile [state]. Displays also the ProfilePicture,
+ * SettingsButton, name and email of th user profile
+ */
+@Composable
+fun ProfileHeader(state: ProfileState, modifier: Modifier = Modifier) {
+  Column(
+      modifier = modifier,
+      horizontalAlignment = Alignment.CenterHorizontally,
+      verticalArrangement = Arrangement.spacedBy(16.dp),
+  ) {
+    ProfilePicture(state)
+    Column {
+      Text(state.name, style = MaterialTheme.typography.h5)
+      Text(state.email, style = MaterialTheme.typography.subtitle2)
+    }
+    SettingsButton(onClick = state::onSettingsClick)
+  }
+}
+
+/** Composes the profile picture given its [state] */
+@Composable
+fun ProfilePicture(
+    state: ProfileState,
+    modifier: Modifier = Modifier,
+) {
+  Box(
+      modifier =
+          modifier.size(118.dp).background(state.backgroundColor.getColorForProfile(), CircleShape),
+      contentAlignment = Alignment.Center,
+  ) {
+    Text(state.emoji, style = MaterialTheme.typography.h3)
+    IconButton(
+        onClick = state::onEditClick,
+        modifier =
+            Modifier.align(Alignment.BottomEnd)
+                .background(MaterialTheme.colors.surface, CircleShape)
+                .border(2.dp, MaterialTheme.colors.primary, CircleShape)
+                .shadow(2.dp, CircleShape)
+                .size(40.dp),
+    ) { Icon(Icons.Default.Edit, "Edit profile icon") }
+  }
+}
+
+/** Composes the settings button */
+@Composable
+fun SettingsButton(
+    onClick: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+  val strings = LocalLocalizedStrings.current
+  OutlinedButton(
+      onClick = onClick,
+      shape = CircleShape,
+      contentPadding = PaddingValues(horizontal = 24.dp, vertical = 12.dp),
+      modifier = modifier) {
+    Icon(Icons.Default.Settings, null)
+    Spacer(modifier = Modifier.width(8.dp))
+    Text(strings.profileSettings)
+  }
+}
+
+/** Composes a Match log using a [title], [subtitle] and an [icon] */
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+fun Match(
+    title: String,
+    subtitle: String,
+    icon: ImageVector,
+    modifier: Modifier = Modifier,
+) {
+  ListItem(
+      modifier = modifier,
+      icon = { Icon(icon, "Match icon") },

```

Comment:
Content description may be null probably.

```suggestion
      icon = { Icon(icon, null) },

```